source.go

```diff
@@ -12,6 +12,150 @@ import (
 	"github.com/pkg/errors"
 )
 
+type FilterFunc = func(string, []string, []string, ...llb.ConstraintsOpt) llb.StateOption
+
+var errNoSourceVariant = fmt.Errorf("no source variant found")
+
+func (src Source) handlesOwnPath() bool {
+	// docker images handle their own path extraction if they have an attached command,
+	// and this information is needed in the case of mounts when we can do path
+	// extraction at mount time
+	return src.DockerImage != nil && src.DockerImage.Cmd != nil
+}
+
+func GetFilter(src Source, forMount bool, opts ...llb.ConstraintsOpt) llb.StateOption {
```

Comment:
This should be uneported, and maybe inlined in `GetSource`?

Reply:
I'd rather not inline it to avoid passing `forMount` through `GetSource`, if that's ok